pkg/meta/model/reorg.go

```diff
@@ -79,8 +79,12 @@ type DDLReorgMeta struct {
 	BatchSize   int64 `json:"batch_size"`
 }
 
-// GetConcurrency gets the concurrency from DDLReorgMeta.
-func (dm *DDLReorgMeta) GetConcurrency() int {
+// GetConcurrency gets the concurrency from DDLReorgMeta,
+// pass the default value in case of the reorg meta coming from old cluster and Concurrency is 0.
+func (dm *DDLReorgMeta) GetConcurrency(defaultVal int) int {
+	if dm == nil || atomic.LoadInt64(&dm.Concurrency) == 0 {
+		return defaultVal
+	}
 	return int(atomic.LoadInt64(&dm.Concurrency))
```

Comment:
Can we run `atomic.LoadInt64()` once?

Reply:
update in https://github.com/pingcap/tidb/pull/57468/commits/7993dcc732854e2e221d796f53a14e9f0de933c5

Reply:
/cc @CbcWestwolf